test/test_fcntl.rs

```diff
@@ -143,3 +146,27 @@ mod linux_android {
         assert_eq!(100, read(fd, &mut buf).unwrap());
     }
 }
+
+#[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd"))]
+#[test]
+fn test_posix_fallocate() {
+    use nix::fcntl::{posix_fallocate};
+    let tmp = NamedTempFile::new().unwrap();
+
+    let fd = tmp.as_raw_fd();
+    let ret = posix_fallocate(fd, 0, 100).unwrap();
+
+    match Errno::result(ret) {
+       Err(Error::Sys(Errno::EINVAL)) => {
```

Comment:
Clippy will get mad about this, a `match` where only 1 arm is checked. Please rewrite this using `if ret.is_some()`, but keep the comment there so it's clear what's going on.

Long-term it'd be better to actually check somehow before the test is run if this will work and skip the test otherwise. Adding a `TODO` here would be helpful for that.